### ISSUE TITLE
ResourceQuota resources don't get pruned

### ISSUE THREAD

**Issue by joekohlsdorf** [Shopify/kubernetes-deploy#259]
Steps to reproduce:
- deploy a ResourceQuota named "compute-resources"
- remove the .yaml and deploy again

Make the observation that the resource still exists:
- `kubectl describe quota compute-resources --namespace=quotatest`

Expected result:
- the ResourceQuota `compute-resources` doesn't exist

**Comment by dturn** [comment]
Thanks for filing this bug. We're aiming to have a fix merged in the next week.

**Comment by joekohlsdorf** [comment]
I took a look at your change and would like to kindly ask you to review the whole list.
`autoscaling/v2beta1/HorizontalPodAutoscaler` for example appears to be missing as well.

**Comment by KnVerey** [comment]
You're right that there are others that are missing. In my opinion we should also prune:
- pod templates
- resource quotas
- service accounts
- replica sets

All of those should be equally simple PRs. As for HPAs, we don't yet have specific support for them: https://github.com/Shopify/kubernetes-deploy/issues/46. That's a different issue, but one that is also a relatively simple change.
